Big User Manual structure revision

ThomDietrich opened the pull request:
## Update, Ready for Review

The open points below are not structure related and can be attended to by their own PRs.
I'll create individual issues for all of them.

---

- [x] Add hr rulers to menu structure
- [x] Rearrange Installation and Setup order
- [x] Rearrange Features articles to Configuration
- [x] Add redirects for all old article locations
- [x] Add placeholders for not yet existing articles
- [x] Insert a link to wiki or whatever suited in new articles (till filled with content)
- [ ] Write Configuration/Overview article with the different ways of configuration (text vs. database, rules files exclusive, sitemap vs. HABPanel, ...)
- [ ] Write PaperUI configuration article
- [ ] Update openHABian article
- [x] Update RaspberryPi article with openHABian details
- [ ] Add SSH disabled info to the RPi article
- [ ] Help: How to post a problem, How to contribute a solution
- [x] Rename menu entry openHAB Designer to Smarthome Designer
- [x] Better Arrangement of "System Integration"
- [x] IFTTT as another "System"
- [ ] ...

-----

### Motivation

I tried to follow these ideas:

**Reducing the menu to the logical parts:**
* Preamble (Introduction, Concepts)
* Prerequisites (Installation and Setup)
* Howto (Configuration and Usage)
* Further possibilities (Addons, Administration)
* Outlook (Help, FAQ, Troubleshooting)

Horizontal rules were introduced as **visual separators** between different topics, without the need to name them or have a collapsing menu for them.

Quite a few articles shifted in their position. All **previous locations** are [set up as redirects](https://raw.githubusercontent.com/openhab/openhab-docs/0c4daa4052341667a057fd740b1de48c4b839f3b/features/items.md). Old links in e.g. the community will not be a problem.

While building the menu, I took care to have a **menu entry for every aspect** of a topic. A [placeholder article](https://github.com/openhab/openhab-docs/blob/0c4daa4052341667a057fd740b1de48c4b839f3b/configuration/transform.md) exists for every article not created yet. The content of these articles is not clear yet. In one place I decided to have two menu entries pointing at the same article, there is room for improvement.

Every article not yet finished (especially placeholders) include a **Contributions Wanted** note with a link to the issue tracker.

------

### Local Environment

If you want to see and test the new arrangement, the probably quickest way is to clone the Repo and execute `vagrant up`. See: https://github.com/ThomDietrich/openhab-docs#alternative-vagrant-vm

ThomDietrich commented:
![grafik](https://cloud.githubusercontent.com/assets/2870104/20863207/af098fe8-b9c3-11e6-971d-777c00c21b44.png)


![grafik](https://cloud.githubusercontent.com/assets/2870104/20863204/9a851560-b9c3-11e6-9de8-916ae9ecea5c.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20863213/c8cd16f2-b9c3-11e6-813a-a2eeb14c02d6.png)

BClark09 commented:
Looking really good so far @ThomDietrich! Did you want people to wait until you've finished the PR for a review? Initially, I notice there's a fair few links to docs.openhab.org directly.

ThomDietrich commented:
@BClark09 I've added a few thoughts to the above description. I'm still changing things around and am trying to play the "Would a drunken ape be able to use this?" game. There are still a few things I'm struggling with and I did not yet touch all aspects.

So, **yes** please review and let me know what you think but please do not expect the perfect result yet. ;)

ThomDietrich commented:
@kaikreuzer I've removed PINE64 from the menu. If you or somebody wants to add content for it I'll happily re-add it ;)

ThomDietrich commented:
I'm currently struggling with the "Configuration & Usage" menu. The idea is to first address the file-based configuration and then the other possibilities. The difference will be described in "Overview" and the horizontal rule is supposed to separate without the need to add list sublevels (which I would like to avoid). 

Do you have an idea? (Ignore "Audio & Video" for now)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20868660/3daa835e-ba61-11e6-9f79-410ab7af74a7.png)

BClark09 commented:
I always thought it would be nice to have an introduction to the benefits of both somewhere and have some sort of selection at the top of the page which talks about both and depending on the selection you can show or hide content. If even possible, this will need a bit of html.

---

_The following page will talk about two different configuration types, [textual] and [through paperUI] make sure you read up on the differences first!:_
  

- [ ] _I want to see how to configure openHAB with text files._
- [ ] _I want to see how to configure openHAB with paperUI_

## Configuration

_blah blah blah_

ThomDietrich commented:
the overview article will talk about this. I already have a (hopefully) nice idea in mind. 
Regarding the separation, I don't think that's a good idea. There is no path 1 and path 2 but rather multiple steps and each can be solved multiple ways, even mixed. I'll try to clarify this once and for all...

ThomDietrich commented:
This is an invite to @kaikreuzer, @rkoshak and @theoweiss to review this (unfinished) PR.
@marziman my.openhab is part of "Installation & Setup", openhab-cloud will be linked there and in the "Addons" section as soon as it's released ;)
@cdjackson I've added a jsondb menu entry to the Administration menu (next to Karaf console, logging, ...). Would you be willing to provide a few words?
@cdjackson There's also a Habmin article under "Configuration & Usage" with the intent to say "You may use Habmin for this and that while defining your home, for more details see ..link..". Any idea what to put in there?
@ghys the same goes for HABPanel...

Thanks for your help!

cdjackson commented:
How are you managing additions into this? Are you planning to merge this, then add PRs for the items you mentioned above, or do you want this PR updated somehow?

kaikreuzer commented:
Thanks @ThomDietrich, I quite like it!

A few comments after a first glance (not ordered by any priority or importance):
- I don't like the "...& Usage" and "Explore the Community" expressions in the menu. I don't have an alternative suggestion, but maybe one of our native speakers (@cdjackson, @rkoshak) has?
- Installation->Raspberry Pi: This is somehow a duplication of "Linux" and "openHABian". Is the idea of this section to list some concrete hardware platforms? If so, I think it is ok to keep it (as many people will specifically search for this), but we should change the page to give the different options right at the top (especially linking to openHABian).
- Besides Synology, we should also list QNAP, see https://github.com/openhab/openhab-qnap-qpkg
- Configuration: HABPanel is not used for system configuration (like Paper UI and HABmin), so I would suggest to remove it from the menu there. It should rather be (only) listed next to Classic UI and Basic UI.
- "Merge Bindings and 1.x Bindings in one table": The more I think about it, the more I'd prefer to keep them separate. There are recurring questions about why OH1 bindings do not turn up in the Paper UI, why there are no Things for them etc. So I think we need to be clear in the documentation that they are configured and used in different ways and I would rather go the opposite way: To split the list of bindings in the Paper UI to have separate tabs for 1.x bindings and OH2-native bindings. WDYT?

ThomDietrich commented:
Hey Kai, thanks ;)

* I'm also not sure about the "Explore the Community" thing. I was looking for something cheerful with the message "If the above didn't provide what you were looking for, check the FAQs, create a thread in the community or contribute". About "Configuration & Usage": I'm open for a better idea. I chose to work with "Word & Word", the message should be "Here you'll find everything related to setting up your own home automation" @rkoshak, @cdjackson

* ... And because of that, I think HABPanel needs a reference (it's only a reference with a few sentences, the main article is still under Addons->UIs) here, because from an end users point of view, HABPanel provides "configuration&usage" capabilities in the form of building the user interface.

* "Installation & Setup" addresses all you need to know to get a openHAB instance running. Hence I'm starting with OS (Linux, Win, Mac), followed by special software solutions (openHABian, docker), followed by special hardware solutions (RPi, Synology, QNAP), followed by more setup steps like SmartHome Designer or the NGINX article.

> Is the idea of this section to list some concrete hardware platforms?

Exactly. In fact one of my main goals (see first posting) was to have "a menu entry for every aspect". I think it's very important to give the interested/unexperienced user a clear view on all relevant topics. The fact that the RPi is a famous platform and that there are a few details in the article made it worth adding/keeping for me...

>but we should change the page to give the different options right at the top (especially linking to openHABian).

Please elaborate :)

* QNAP: check!

* Merge: The table would of course state the "type" of a biding as a column. I think this is the important step, also for PaperUI, where you are having problems because of the lack of this information in one combined table - which is quite understandable... What do the others think?

ThomDietrich commented:
@cdjackson the PR will be merged as soon as we agreed on the important changes. Till then I would suggest to communicate via comments here. You could provide a short introduction on jsondb here now - which I will add - and create a full article as PR later... agreed?

ghys commented:
Thanks @ThomDietrich for doing this 👍 

Regarding HABPanel - I've been willing for a long time to provide some sort of documentation for the more advanced features (you might be aware there are ways to code your own UI parts, but there is a learning curve: you have to learn a little bit of AngularJS/HTML/CSS with some custom specifics - exposed methods, widget models, and so on).
Right now most of the knowledge about this stuff is in community.openhab.org. I don't know how relevant it would be to include this stuff in the official documentation...?

kaikreuzer commented:
> And because of that, I think HABPanel needs a reference

Ok, got it :-)

> Please elaborate :)

At the moment, the RPi page explains how to set it all up manually. People might do all of that and find out about openHABian only afterwards - and thus have done a lot of stuff manually, which would have come for free with openHABian. That's why I mean that there should be at least a note pretty high up on that page.

> The table would of course state the "type" of a biding as a column. 

I don't really think that an extra column is a good way to express the huge differences. Note that for the "simple" packaging, I already filter out all OH1 bindings in Paper UI as they do not make any sense for a UI-driven setup. As @rkoshak suggested, having one menu entry per tab in the Paper UI would give the best recognition value - and I think splitting it there makes sense in any case.

kaikreuzer commented:
@ghys: I think the intention of this chapter is purely end-user documentation - definitely NOT how to "code your own UI parts". Rather how to create a dashboard, add widgets and configure them.

ghys commented:
okay, understood :)

BClark09 commented:
I don't have any comments to add to, this is excellent @ThomDietrich! 

Instead of "Explore the community" why not "The openHAB community"? IMO community is already a friendly and inviting word. 

I don't mind "Configuration & Usage", although when I think about it, I have always automaticlly clicked "Configuration" in the current docs expecting to see the information about items etc. only to be greeted by my own NGINX page. Why not keep this as simply "Configuration"?

kaikreuzer commented:
> Why not drop the usage and keep this as simply "Configuration"?

👍 . Sorry that I missed you to mention as another native speaker ;-)

ThomDietrich commented:
@kaikreuzer RPi: ah got it! Yes I'll do that. There still much to do. I've added a point in the todo list on the top.

@kaikreuzer @rkoshak Bindings: Okay I'll leave this section untouched for now. Let's discuss this topic in the other thread.

@ghys @kaikreuzer HABPanel: but still the mentioned documentation would be nice to have in articles of some kind. From my point of view well organized information in the community is acceptable as well. Check the Tutorials & Examples section to get an idea how you could continue. [Example](https://community.openhab.org/t/influxdb-grafana-persistence-and-graphing/13761).
Regarding the text in question here: Just a simple few introductional lines provided here in the comments would be enough ;)

@BClark09 Grey: I know what you mean. I was experimenting for a short time only. Will try lighter grey.

@BClark09 @kaikreuzer "The openHAB community" is also not quite the thing either. It should be something that says "Go here for help, the forum, the FAQs, Tutorials, Examples, HABPanel widgets ans so on..."

@BClark09 @kaikreuzer "Configuration": I have to agree, that happened to me to. ;) Configuration is definitely the right name, I wanted to add a second part to 1. better describe the content and especially address the whole "Bring your own smart home to live, Create your own home automation solution" part rather than just saying "Configure your preferred three settings". Get me?

BClark09 commented:
"Community Guidance" then? We're limited on space but the phrase seems to say "come to for guidance" as well as "go here to guide others". 

I get ya, and as I said I don't think "...& Usage", is a problem, but it does seem a bit superfluous on a docs website and can also apply to parts of Administration and Addons.

rkoshak commented:
I've only a few minutes so am going to make a few quick suggestions and comments, mainly based on the screen shots and the comments already made. I'll check it out and review it more detailed later today. 

- I like what I see a lot

- I think that moving towards single work nouns for the top level menu entries where possible is a good idea. So I would drop the " & Usage" from the Configuration and remove the "& Setup" from the Installation. Also, the difference between "Configuration" and "Setup" may not be clear in a lot of people's mind and by cutting those words out is becomes much more clear where to look.

-  "Explore the Community" could be simplified to just "Community" and not lose any meaning

- "Installation->Raspberry Pi" I agree with @kaikreuzer, we shouldn't duplicate the same Linux content here. However, there are considerations one need make when using ARM based board computers like the Pi. Maybe it makes sense to make this a "Board Computers" menu that references the Linux installation instructions and limit the content to just special capabilities (e.g. openHABian) and concerns (SD card corruption). I'm not sure if the title "Board Computers" is a little too arcane though, but I see this as a place to look for those with any of the Pi computers, Pine64, CHIP, BeagleBone, etc.

- If the lists are separated in PaperUI I'm more OK with having them separated in the docs, as long as the docs mirror the organization in PaperUI. If we are pushing PaperUI as the default admin UI, then the docs and the UI must mirror each other as much as possible. That will greatly help users intuit where to look for information without needing to search.

However, I want to be careful we are not solving one problem (confusion about differences between 1.x and 2.0 bindings) with another one (how do I find the docs for my binding?), particularly for those that have both versions. I don't have a good solution. If we put them all together the distinction may be lost yet if we completely separate them the fact that there are two different ones may be missed. It also makes the job harder for those users who just want to know whether OH supports technology X. Now they have two places they have to look, assuming they understand they need to look both places.

No matter what we must make the distinction between 1.x and 2.0 clear no how matter it is presented. I think I still lean towards merging the tables and just making it very clear in the table which is which, but I can be persuaded that separate tables is a good way to go too.

If we do merge the tables, what I'd like to see are two very distinct icons (like a blue circle with a 1.x in it for 1.x and a green circle with a 2.0), one for each version and I'd like to see that icon also appear in PaperUI. Right now, the much smaller and muted gray text that shows the version in PaperUI belies the fact that the version is very important.

- Speaking of search, and this is a discussion for another issue I'm sure, but having a search function sure would be nice, particularly to search for add-ons.

>HABPanel provides "configuration&usage" capabilities in the form of building the user interface.

I still think it is confusing to keep it with PaperUI and Habmin, even with just a little blurb and a link to the main page. For all intents and purposes HABpanel is a user's UI. The fact that is provides a built in mechanism to create the panel does not raise it to the level of an admin UI.

>The fact that the RPi is a famous platform and that there are a few details in the article made it worth adding/keeping for me...

And that is a good argument I even made for myself when I made my comment above. The Raspberry Pi is the most famous board computer and the one that most users who are just coming to this sort of computer will be most familiar with.

- @ghys, I think a tutorial page or pages that walk the user through "just enough" of each topic needed to get up and running and custom configured along with separate and easy to find tutorials for more advanced things in the forum would provide a happy medium.

- While we are talking about UIs, what about CometVisu?

As soon as I can I'll pull down the code and review it more closely.

kaikreuzer commented:
> It also makes the job harder for those users who just want to know whether OH supports technology X.

This brings me to http://www.openhab.org/features/supported-technologies.html.
I would actually like to keep some "feature overview" on openhab.org itself, which is more marketing oriented than user documentation - but if we keep this, we would have this single list, where newbies can check whether openHAB might be at all interesting to them.

Regarding separate tables on docs.openhab.org: I think for us openHAB 1.x users, the 1.x bindings are important and we know how to deal with them. But I think for all newbies, the 2.x bindings are much preferred and when people learn how to setup and configure stuff, it should be enough for most to understand how to do it for 2.x bindings, so I'd really prefer to treat 1.x bindings as something like "if you are an OH1 user or you really need that stuff, have a look here as well". As mentioned, for the "simple" package, I would completely keep them out of scope.
With more and more people moving over to OH2, I think there will be less and less reason to focus on 1.x bindings, so we should keep that in mind for the documentation.

ThomDietrich commented:
I didn't read @rkoshak's novel yet ;P I'll come to that later. Just a short comment on @kaikreuzer:

> But I think for all newbies, the 2.x bindings are much preferred

I'm not sure about that. You do not select addons but rather your environment dictates them. If I am a fond enocean and kodi user, I'll have a look at these bindings, whatever the "classification I can not relate to but seems to be something I should look out for".

rkoshak commented:
> With more and more people moving over to OH2, I think there will be less and less reason to focus on 1.x bindings, so we should keep that in mind for the documentation.

Agreed except all persistence add-ons are 1.9, all transformations are 1.9, all actions are 1.9, and right now the bulk of bindings only have a 1.9 version. Newbies need to care about these as well. I don't think we are there yet where we can treat the 1.9 stuff as "here is some other stuff you might be interested in too."

I do like the marketing oriented documentation though. It does help address some of my concerns.

As an end game though I completely agree, the 1.9 bindings and add-ons will eventually need to be marginalized in the docs with the main focus being on the 2.0 stuff. I just think that is premature at this point.

However, if we are only talking about bindings because the other stuff will be moved to their own pages then I'm closer to being OK with this.

kaikreuzer commented:
> all transformations are 1.9

No, all are 2.x.

> right now the bulk of bindings only have a 1.9 version

Definitely not anymore as soon as [all this](https://github.com/openhab/openhab2-addons/pulls?q=is%3Apr+is%3Aopen+label%3Anewbinding) is merged.

kaikreuzer commented:
> all actions are 1.9

This is another good example why it makes sense to keep things separate. We will soon have a new category "rule modules" for the new rule engine and it would not make sense to mix those two categories up, although they serve a similar purpose.

This is similar to my point about the bindings: People need to work in a pretty different way with them wrt setup and configuration and we anyhow have to explain this - and whoever can avoid having to understand BOTH ways should imho do so.

kaikreuzer commented:
> If I am a fond enocean and kodi user

If there are many of them, there will likely be a developer who ports the binding to OH2 - so having to use 1.x is a temporary measure and rather for the extravagant (i.e. not broadly used) bindings.

ThomDietrich commented:
This is getting out of hand. :open_hands: Let's move the whole add-ons/bindings discussion to https://github.com/openhab/openhab-docs/issues/145 Agreed?

Everything else said so far: Thanks guys, I'll look into the needed changes and get back to you as soon as the Commit is in. In the meantime You may continue reviewing ;)

ThomDietrich commented:
Update: I have worked on your suggestions.

**RPi:** the Raspberry Pi article now first points to the openHABian article and then gives some manual setup suggestions (as known from before).

**Synology:** I have updated the article. @cniweb the versions you linked below are not up to date anymore. Is mentioning specific version numbers in the more static docs article really the way to go?

**QNAP:** I've added a short article. @thopiekar could you PLEASE provide a short suggestion on what to put in there!?

**Menu Names:** Hope you like it better that way.

**Many small changes:** can't name them. Please check the commit history

ThomDietrich commented:
![grafik](https://cloud.githubusercontent.com/assets/2870104/20943041/3e5f8f6c-bbfe-11e6-8bad-ef7ec9a62a5b.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20943052/49baf194-bbfe-11e6-8e27-09f80cc5d3c0.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20943069/5525040c-bbfe-11e6-83cb-4642395a67ec.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20943083/65305658-bbfe-11e6-9642-3076a16d01f0.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20943098/73ea4500-bbfe-11e6-9816-35555f057ae6.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20943274/0cbcece2-bbff-11e6-9cc3-55cd838c1b50.png)

![grafik](https://cloud.githubusercontent.com/assets/2870104/20943283/15516a2c-bbff-11e6-8b3a-ea27c611a027.png)

ThomDietrich commented:
Everyone: Please take the time to review these changes. Your input would be great at this point...

ThomDietrich commented:
@kaikreuzer I played around with the bindings and addons a little bit more. I do not want to restart the "one table for all bindings" discussion. I've broken down the 1.x addons table to three tables for actions, persistence and bindings:

![grafik](https://cloud.githubusercontent.com/assets/2870104/20990965/0fdc4f5c-bcdc-11e6-95b8-dd6834d04af4.png)

Please decide how to finalize this (later changes can still do more):

* revert
* leave it this way
* put 1.x bindings in their own menu entry, resulting in "2.0 Bindings, 1.x Bindings, 1.x Addons"
* put 1.x bindings inside 2.0 bindings article as extra table (not merged in any way), resulting in menu entries "Bindings, Addons"

rkoshak commented:
I like having the Bindings, Actions, and Persistence separated like this. It makes it much easier to find.  No matter how all of this gets organized I would try to keep them separated.

Everything else I've seen looks good to me. I think this is a great improvement.

ThomDietrich commented:
@rkoshak thanks. Which option would be your preferred one then?

kaikreuzer commented:
> put 1.x bindings inside 2.0 bindings article as extra table (not merged in any way)

Maybe this is a good compromise, which would leave us with a single "Bindings" menu entry, but still be able to clearly show the "full support" bindings separated from 1.x bindings.

I'd agree with @rkoshak that then the "1.x Addons" page should be split into new menu entries for "Actions" and "Persistence", so that this is in line with the tabs on the Paper UI. Does that make sense?

ThomDietrich commented:
Absolutely. This was also my preferred choice ;) Btw. I've already added a few sentences to address the ["full support" bindings](https://github.com/ThomDietrich/openhab-docs/blob/3227576419b360f73a15ab33ea1aa4f496f67734/addons/bindings.md). I will do the needed changes shortly.

**Other noteworthy changes:**

* @kaikreuzer I've added IFTTT as a menu entry under 3rd Party Integration
* @rkoshak I've updated the migration tutorial to be linked under "migration.html" to follow the behaviour of the rest of the page. Doesn't seem to have any unwanted effects.

**Open Issues:**

* I would still be happy to get support in the form of short articles from @thopiekar @cniweb @cdjackson @ghys as said above. This is just a reminder, sorry for being pushy. Please just let me know if you do not have the time currently.
* Configuration -> PaperUI: I've problems to define/write something here as I am still not using PaperUI (with the latest changes, I might soon). @rkoshak you wanna have a go at it? I just need a few short introductional lines and a link to the PaperUI readme.
* Configuration Overview article is still missing (not mandatory for this PR)
* openHABian article is at it's minimum content (not mandatory for this PR)

BClark09 commented:
I'm all for these changes, although I have a couple of points:

1. I think the contribution card would look better with and Orange bolded title: (#FF6600) and the same colour hyperlink as in the navigation bar (#039BE5):

 ![selection_001](https://cloud.githubusercontent.com/assets/8983584/21010238/e417fdd2-bd42-11e6-8bfc-b045ada4022f.png)

2. I also compiled a list of links broken as part of this update (I think): [siteerrors.txt](https://github.com/openhab/openhab-docs/files/639541/siteerrors.txt)

ThomDietrich commented:
Thank you @BClark09 

1. I know right. To be honest, I didn't want to mess around in a css file for that :) But okay I'm on it.
2. Nice, will look into it.

thopiekar commented:
Hey @ThomDietrich, there is not much I can tell about QNAP. I've added a simple howto into my repository, see https://github.com/openhab/openhab-qnap-qpkg .
Feel free to take the photos plus instructions and move them into your pages.

rkoshak commented:
> @rkoshak thanks. Which option would be your preferred one then?

If I were king for a day I'd make the Add-ons folder mirror the tabs in PaperUI:

- Bindings (with OH 2 at the top, OH 1.x at the bottom in a separate table)
- User Interfaces
- Persistence
- Actions
- Transformations
- Voice
- Misc

The more closely the docs mirror the "default" UI the better.

Second choice would be your option 3 with them on the same page but in separate tables.

My biggest concern is that we are treating the categories of the add-ons inconsistently. For example, why separate the UI and voice add-ons but bury persistence and actions inside some other menu item that does not make clear that is where one should look. For a newbie, there is little about the term "add-on" that tells me I should look at Bindings and Add-ons to learn how to integrate with InfluxDB.

ThomDietrich commented:
@kaikreuzer @rkoshak I checked the OH1 wiki and found articles to these additional 3rd party system integrations:

* Chrome Addon: https://chrome.google.com/webstore/detail/mini-openhab-controller/lphilmckagfijcikbfoibgekcoodaobd
* Slack Integration: https://github.com/openhab/openhab/wiki/Slack-integration Possible error: https://community.openhab.org/t/slack-integration-how-to-integrate/17943
* I also want to add an article on the possibility to make your own application interact via REST interface

And more: 
![2016-12-09 12_40_19-application integration openhab_openhab wiki - pale moon](https://cloud.githubusercontent.com/assets/2870104/21047906/b5887198-be0c-11e6-86bf-30bbe2d4ad55.png)

Any preference on what to include in docs.openhab.org?

kaikreuzer commented:
> Any preference on what to include in docs.openhab.org?

Imho we should only include the stuff which is "officially" supported by openHAB, i.e. all the IO/Misc integration bundles.

All the rest should be left somewhere outside the official documentation as it usually requires additional software from somewhere else to be set up and it is often rather a hack than a feature.
Nonetheless it is of course interesting for some. But as it can easily become a huge list, I feel that the best place is probably https://community.openhab.org/c/setup-configuration-and-use/tutorials-examples (or a new category). Wdyt?

rkoshak commented:
I agree, the list will grow large and difficult to maintain and the forum is a good place for these. 

I really like the idea of a page that explains how to create Items/Rules to interact with third party APIs. Perhaps some discussion about how to do this using exec binding as well as REST could be useful to some. The focus would not be on the binding details itself and be more of an approach to translate a documented API into OH.

ThomDietrich commented:
Everyone, especially @kaikreuzer, @BClark09, @rkoshak,
I've invested a few more hours into many details, bugs and improvements. You are best off just reading the commit messages.

This could go on and on but in terms of big structural changes, I've reached a point where we could just wrap this up. There is still a lot to do but all todos are not structure related and I think it's better to handle these in individual issues/PRs.

Please use a local environment (Vagrant) to look at the results. The reviews may begin!

ThomDietrich commented:
Quick note: I have reverted one single commit causing 13,213 deletions. These were clouding the the whole PR, hopefully it's now easier to review. I can still pick the commit for an independent PR later.

I'll not change anything from now on (except after your review comments).

kaikreuzer commented on _config.yml:
```diff
@@ -1,6 +1,6 @@
 # Site settings
-title: openHAB
-description: Open Source Smart Home Solution
+title: openHAB 2.0
```
I would suggestion to only talk about openHAB 2 everywhere from now on. The closer we come to an 2.0 release, the sooner 2.1 will be on the horizon and we won't want to update everything again then!

ThomDietrich replied:
Fair point. I'll do a global search and des.. replace.

kaikreuzer commented on _includes/user-menu.html:
```diff
@@ -1,101 +1,130 @@
+      <hr />
+      <li><a href="{{docu}}/configuration/packages.html">Initial Setup Packages</a></li>
+      <li><a href="{{docu}}/configuration/ui-based.html">Paper UI Discovery</a></li>
+      <li><a href="{{docu}}/configuration/ui-based.html">Paper UI Configuration</a></li>
```
having 4 entries with the same url results in all of them being selected in the menu:

<img width="206" alt="screen shot 2016-12-12 at 15 22 26" src="https://cloud.githubusercontent.com/assets/3244965/21102736/ad4b30a4-c07f-11e6-9327-fe865a74e2ef.png">

Is this intended?

ThomDietrich replied:
No it's not. This is the last remaining construction site in my current scope. This should be settled before merging. I was asking for someone to help provide a meaningful menu structure and content. I am not toooo familiar with Paper UI.
I would like to address the "new" configuration possibilities in this menu part. Paper UI provides multiple of these, including initial setup and the experimental Rule Engine. Which menu entries would you like to see? I'll add one document per menu entry. Before HABmin I'll add another `<hr />`.

Btw. "Next Gen Rules"/"Experimental Rule Engine" needs a catchy name. Do you have something in mind?

kaikreuzer replied:
> Which menu entries would you like to see? I'll add one document per menu entry.

As long as we do not have much content, a simply menu entry for the Paper UI might be enough - after all, we also have discovery/items/things/rules in HABmin and this also only has a single menu entry :-)
Also, I wonder how much of the content should better be documented at ESH itself and only included here. I know, it makes it all more difficult, but after all, ESH needs the more or less the same content.

> Btw. "Next Gen Rules"/"Experimental Rule Engine" needs a catchy name. Do you have something in mind?

Maybe just call it "Rules (Experimental)" for now. I would hope that we some time manage to merge https://github.com/eclipse/smarthome/pull/1783, which will then make the rule engine also a replacement for the DSL rules, so we will be left with a single rule infrastructure again, which will make documentation easier as well again.

ThomDietrich replied:
@ghys I've removed HABPanel from the Configuration menu. It's still available under Add-ons > UIs. You didn't provide any content for the article and with the changes here it's not a really a good match anymore.

Anyway. If you decide to write a Configuration oriented article for HABPanel, as it's own PR, that would be great.

ThomDietrich commented:
Thanks Kai! I'm happy to contribute my part to this great project.

kaikreuzer commented:
Btw, one thing I wondered: Shouldn't we also add pages for Android and iOS apps under UIs? They are pretty important for almost all users and if docs.openhab.org becomes our main source of wisdom, we should not forget about them. Wdyt?

kaikreuzer commented on _includes/user-menu.html:
```diff
@@ -1,101 +1,130 @@
-          <li><a href="{{docu}}/addons/io/hueemulation/readme.html">Amazon Echo</a></li>
+          <li><a href="{{docu}}/addons/io/hueemulation/readme.html">Hue Emulation</a></li>
           <li><a href="{{docu}}/addons/io/imperihome/readme.html">ImperiHome</a></li>
+          <li><a href="https://my.openhab.org/docs">my.openHAB (external)</a></li>
```
change to `{{docu}}/addons/io/openhabcloud/readme.html` and label to "openHAB Cloud"

kaikreuzer commented on _includes/user-menu.html:
```diff
@@ -1,101 +1,130 @@
+          <li><a href="{{docu}}/configuration/restdocs.html">openHAB REST API</a></li>
+          <li><a href="{{docu}}/configuration/rules-ng.html">Rule Engine (Experimental)</a></li>
+          <hr />
+          <li><a href="https://my.openhab.org/docs/ifttt">IFTTT (external)</a></li>
```
change link to `http://www.myopenhab.org` (I'll add content about IFTTT soon) and the label to simply "IFTTT".

kaikreuzer commented:
@ThomDietrich & all, do you see a chance to merge this by tomorrow morning? I plan to do the myopenHAB announcement and I expect quite some people to visit the documentation page (and I would like to directly link to the openhab-cloud bundle documentation in it). Would be perfect :-)

ThomDietrich commented:
Hey Kai, I have a christmas party this evening.So I did not plan to attend to your comments today but it's doable. I'll finish everything up till 18:00 (UTC+1).
@BClark09 @rkoshak @cdjackson any last minute comments? :)

BClark09 commented:
On the latest version I've only had a very quick read but it looks very good to me! I think it would be safe to merge and fix any minor errors later.

rkoshak commented:
I haven't had time to do a detailed review but the high level structure and content looks good to me.

ThomDietrich commented:
@kaikreuzer I have added menu entries and articles with little content for the Android and iOS apps under Addons>UIs.
@digitaldan please contribute some content for these articles. A feature list, two screenshots, whatever you think is needed ;) Thanks.

kaikreuzer commented:
Thanks! So will you merge or do you want me to do that?

ThomDietrich commented:
One sec, I'm not finished yet. Let me do the last few things....

ThomDietrich commented:
That's it. 40 commits and I'm finally finished :smiley: :tada:

Thanks everyone for your comments. The work is sadly not over. I have marked a lot of articles with "Contribution Wanted!", meaning, that they are either empty, incomplete or outdated. I'll open individual issues for all of these in the next couple of days and sincerely hope to find people willing to write.

@kaikreuzer go for it. The Glühwein is waiting for me ;)

kaikreuzer commented:
Enjoy your party, you have deserved it!!! 🎉
